api/build.gradle

```diff
@@ -485,47 +476,23 @@ dependencies {
   // Dependencies for Swagger codegen. This should include all dependencies required by Swagger's
   // default okhttp API codegen templates (see https://github.com/swagger-api/swagger-codegen/blob/v2.2.3/samples/client/petstore/spring-stubs/pom.xml)
   // plus any custom templates included in our repo under the swaggerTemplateDir.
-  generatedCompile "org.springframework.boot:spring-boot-starter-data-rest:${SPRING_BOOT_VERSION}"
-  generatedCompile "io.springfox:springfox-swagger2:${SPRINGFOX_VERSION}"
-  generatedCompile "io.springfox:springfox-swagger-ui:${SPRINGFOX_VERSION}"
-  generatedCompile "com.squareup.okhttp:okhttp:${OKHTTP_VERSION}"
-  generatedCompile "com.squareup.okhttp:logging-interceptor:${OKHTTP_VERSION}"
-  generatedCompile "com.google.code.gson:gson:${GSON_VERSION}"
-  generatedCompile "joda-time:joda-time:${JODA_VERSION}"
-  generatedCompile "com.fasterxml.jackson.datatype:jackson-datatype-joda:${JACKSON_VERSION}"
+  jvmCodegenDependencies "org.springframework.boot:spring-boot-starter-data-rest:${SPRING_BOOT_VERSION}"
```

Comment:
Why the rename? Is this fundamentally different from the other targets we have (which follow a consistent naming pattern?) - `compile`, `testCompile`, `toolsCompile`

Reply:
I was confused by both the "compile" part and the past tense of "generated". I thought from the comment all these were just for the codegen step. I'm thinking of compilation narrowly as going from source to bytecode; is there a broader sense I should be thinking of?

I filed https://precisionmedicineinitiative.atlassian.net/browse/RW-4577 to pull the codegen stuff into its own gradle file and tidy up the remnants so this is all easier to navigate.

Reply:
No, it's going from source to bytecode. These are the dependencies required to compile the generated Java code, i.e. the generated source files contain imports of these 3p libraries. The codegen tooling is not something we're compiling ourselves, we're just using a gradle plugin to run that.